lightwood/mixers/helpers/default_net.py

```diff
@@ -34,11 +36,21 @@ def __init__(self, dynamic_parameters,
         super(DefaultNet, self).__init__()
 
         if shape is None and pretrained_net is None:
-            shape = [self.input_size, max([self.input_size*2,self.output_size*2,400]), self.output_size]
+            # default shape, inflated, yields 240k params if input has 200 components
+            shape = [self.input_size, max([self.input_size*2, self.output_size*2, 400]), self.output_size]
+
+            # if NN is too big, we do not inflate and aim for max target_params
+            if reduce(lambda x, y: x*y, shape) > target_params:
```

Comment:
Every time I can't think of an imperative equivalent simpler than the usage of a HOF the functional programming old gods gather a bit more strength.